language/move-lang/src/parser/ast.rs

```diff
@@ -380,6 +381,10 @@ pub enum Exp_ {
     Copy(Var),
     // n
     Name(Name),
+
+    // [m::]n<t1, .., tn>
+    GenericName(ModuleAccess, Option<Vec<Type>>), // spec only
```

Comment:
Why is this a separate node? Why can't we just merge it with `Name`?

My guess is precedence reasons?

Reply:
It's only supported for spec lang, so I wanted to keep it separate from the rest of the AST. If we merge this, then when working the expansion AST, we would have to analyze the nodes content (only simple name in module access, no generics), whereas right now, we can just reject it directly via syntax.  I don't mind that much, but it appeared clearer this way.

Reply:
Done. (See also below.)